### ISSUE TITLE
Add comparison function to DataFieldOptions and DataField

### ISSUE THREAD

**Issue by pafvel** [foundryvtt/foundryvtt#11451]
### User Experience

In order to compare and sort values that has application specific meaning. For example dice stored as strings will sort in the wrong order when using regular string comparison: "D10" < "D8". The comparison function should probably have the same properties as the one used in Array.sort().

In my specific case I am changing the css style on values that have been affected by active effects. The code uses the datamodel to retrieve values and then compares and highlights depending on if it's an increase or a decrease. It is elegant until the code needs to check if the field name is one of the fields that are storing dice, in which case custom comparison needs to be invoked. Using the DataField for comparison would have made the code agnostic of field names.

**Comment by JPMeehan** [comment]
Couldn't you leverage v12's `applyChange` and related methods?

**Comment by pafvel** [comment]
No. I want to compare values, not apply an active effect. In my example use case, the ae has already been applied and the comparison happens when rendering an actor sheet.
